polaris-react/src/styles/foundation/_shadow-bevel.scss

```diff
@@ -0,0 +1,30 @@
+/// Applies a shadow-bevel effect to an element.
+/// @param $boxShadow - Polaris ShadowBorderAlias custom property.
+/// @param $borderRadius - Polaris BorderRadiusScale custom property.
+/// @param $content - Declaration passed to the pseudo-element. Used for toggling the bevel effect.
+/// @param $zIndex - Declaration passed to the pseudo-element. Used for adjusting the stacking order.
+@mixin shadow-bevel(
```

Comment:
Do we want this as a sharable mixin or could we only use the `ShadowBevel` component as our preferred method of style composition?

I'm thinking we could still have the mixin but keep it local to only the `ShadowBevel.scss` so other components are more likely to use the `ShadowBevel` component instead. Wdyt? @aaronccasanova

Reply:
> Do we want this as a sharable mixin or could we only use the `ShadowBevel` component as our preferred method of style composition?

I think having only one solution could introduce friction to the upgrade path. I mentioned in the PR description that the mixin is intended for use when access to component markup is limited or would require significant refactoring. For example, have a look through the hardcoded [`--p-font-*` custom properties](https://github.com/search?q=repo%3AShopify%2Fpolaris+path%3Apolaris-react+--p-font-&type=code) that can't easily upgrade to the `Text` component, but would benefit from this same strategy (e.g. `@include text-heading-sm`, `@include text-body-lg`, etc). Additionally, I would personally opt for the mixin before the component to [help reduce excessive DOM size](https://developer.chrome.com/en/docs/lighthouse/performance/dom-size/) costs.

> I'm thinking we could still have the mixin but keep it local to only the `ShadowBevel.scss` so other components are more likely to use the ShadowBevel component instead. Wdyt? @aaronccasanova

I also think this is appropriately positioned in the library. The effect is foundational to the system and it is customary to import these types of primitives from `styles/common`

Reply:
Sounds good to me! Thanks for the added context, examples, and rational 👍